### DISCUSSION TITLE
Trying to build  with `-fwasm-exceptions` is failed with an error about undefined  __cxa_find_matching_catch_2

### DISCUSSION THREAD

**Discussion by aharondavid** [emscripten-core/emscripten#21916] [Q&A]
emcc 3.1.55.
Windows 11.

Trying to build  with `-fwasm-exceptions` is failed with the following error:
`error : undefined symbol: __cxa_find_matching_catch_2 (referenced by root reference (e.g. compiled C/C++ code))`

Setting the `-fwasm-exceptions` is follow by the next steps: 

1. My project is built by one static `wasm` file that linking with several libs.
2. first, I set the flag `-fwasm-exceptions`  to all my source\lib in my project.
3. verify that there is no additional used in `-fno-exceptions`` -fyes-exceptions`
4. linking with `-fwasm-exceptions`

**result:**
get linking error:
`error : undefined symbol: __cxa_find_matching_catch_2 (referenced by root reference (e.g. compiled C/C++ code))`

**The build command:**

`emcc.bat  Emscripten/Debug/TEApiWrapper81_for_emscripten.o Emscripten/Debug/Main.o --pre-js pre-js.js -ljpeg.Emscripten -lxml2 -lprjplg.Emscripten -lproj -lexpat -lgdal -lgeos -lgeos_c -l3ds.Emscripten -lTerraExplorerX.Emscripten -lCore.Emscripten -l7zip.Emscripten -lzip.Emscripten -llibzip.Emscripten -lgisplg.Emscripten -lMptDll.Emscripten -lOpenGL.Emscripten -lPlgShell.Emscripten -lSLLogger.Emscripten -lSLLoggerLib.Emscripten -lSLStrings.Emscripten -lTerraCommon.Emscripten -lTerra.Emscripten -lTESWQ.Emscripten -lWFSPlugin.Emscripten -lOGRPlugin.Emscripten -lsl_png.Emscripten -ldraco.Emscripten -lOpenCTM.Emscripten -lwebp.Emscripten -lMPTPlg.Emscripten -lOGRSpatialReference244.Emscripten -s ALLOW_TABLE_GROWTH -s WASM_BIGINT  -s EXPORTED_RUNTIME_METHODS=ccall,cwrap,addFunction,allocateUTF8   -s EXPORTED_FUNCTIONS=_main   --bind -s USE_PTHREADS -s ALLOW_MEMORY_GROWTH=1    -s ASSERTIONS  -s FETCH   -s TEXTDECODER=0  -lidbfs.js -s PTHREAD_POOL_SIZE=GetThreadPoolSize()  -s REVERSE_DEPS=all -s STACK_SIZE=5mb -s DEFAULT_PTHREAD_STACK_SIZE=5mb -s ALLOW_BLOCKING_ON_MAIN_THREAD=1  -s PTHREAD_POOL_SIZE_STRICT=2  -fwasm-exceptions -O0 --preload-file com.skyline.terraexplorer -o C:/Projects/TerraExplorer_v8_0_0_Dev/SLSource/Emscripten/Debug/TEF/TerraExplorerApp.Emscripten.js -LC:/Projects/TerraExplorer_v8_0_0_Dev/SLSource/Emscripten/Debug/ -LC:/Projects/TerraExplorer_v8_0_0_Dev/SLSource/Emscripten/TerraExplorerApp.Emscripten/libs`

**The output result:**

```
undefined symbol: __cxa_find_matching_catch_2 (referenced by root reference (e.g. compiled C/C++ code))
warning: To disable errors for undefined symbols use `-sERROR_ON_UNDEFINED_SYMBOLS=0`
warning: ___cxa_find_matching_catch_2 may need to be added to EXPORTED_FUNCTIONS if it arrives from a system library
error: undefined symbol: __cxa_find_matching_catch_3 (referenced by root reference (e.g. compiled C/C++ code))
warning: ___cxa_find_matching_catch_3 may need to be added to EXPORTED_FUNCTIONS if it arrives from a system library
error: undefined symbol: __cxa_find_matching_catch_4 (referenced by root reference (e.g. compiled C/C++ code))
warning: ___cxa_find_matching_catch_4 may need to be added to EXPORTED_FUNCTIONS if it arrives from a system library
error: undefined symbol: __resumeException (referenced by root reference (e.g. compiled C/C++ code))
warning: ___resumeException may need to be added to EXPORTED_FUNCTIONS if it arrives from a system library
error: undefined symbol: llvm_eh_typeid_for (referenced by root reference (e.g. compiled C/C++ code))
warning: _llvm_eh_typeid_for may need to be added to EXPORTED_FUNCTIONS if it arrives from a system library
warning: JS library symbol '$allocateUTF8' is deprecated. Please open a bug if you have a continuing need for this symbol [-Wdeprecated]
```

**The attach full_log.csv file is contain the output build after turning on the EMCC_DEBUG:**
[full_log.csv](https://github.com/emscripten-core/emscripten/files/15260229/full_log.csv)

**Comment by sbc100** [comment]
`__cxa_find_matching_catch_NN` symbols come from object file that were built without `-fwasm-exceptions`.   Can you do a clean build and be sure to build all of your object files and libraries with that flag?

You may be able to add `-Wl,--trace-symbol=__cxa_find_matching_catch_2` to your link flags to find out exactly which object file references that symbol.

**Comment by aharondavid** [comment]
You right - there was one thread party lib that built with `emconfigure` but without the  `-fwasm-exceptions`.

Thanks!
